### ISSUE TITLE
Named windows

### ISSUE THREAD

**Issue by Ram-Z** [The-Compiler/qutebrowser#1096]
* [ui]window-title-format
  * Add {name} field
  * Possibly add {name_sep} akin to {title_sep}
  * Change the default to {perc}{title}{title_sep}qutebrowser{name_sep}{name}

* Commands
  * `:open -w -n <window-name>`: opens a new window the name `<window-name>`
  * `:set-window-name <window-name>`: set the name of the current window and updates window title
  * Any other command that could open a new window should probably get the --name option as well (`:help`, etc)

* Session
  * Save the window name into the session files

**Comment by Ram-Z** [comment]
I've started work on this, will post WIP PR soon.

I can't seem to register a `str` option to the [`:open`](https://github.com/The-Compiler/qutebrowser/blob/master/qutebrowser/browser/commands.py#L279) command. Adding another bool with `openurl(self, url=None, b=True, [...])` works, but `openurl(self, url=None, s="", [...])` just doesn't generate the `-s` option.

**Comment by The-Compiler** [comment]
Positional arguments are added as positional arguments to the qutebrowser command as well, unless they are a boolean toggle.

You'll need to make it a keyword only argument by doing `openurl(self, url=None, [...], *, s="")`.

**Comment by The-Compiler** [comment]
FWIW, I disagree with adding a window name option for everything opening a new tab. `:open` is fine, for everything else it sounds too specialized. Note how `:help` doesn't have `-t`/`-w`/`-b` either :wink:

**Comment by Ram-Z** [comment]
Agreed, I wasn't too fond of that either, just some thoughts I wanted to write down before going to sleep.

**Comment by amerlyq** [comment]
@Ram-Z, how is progress on this feature? Will it allow to open links in specified named window from command line? Like I wanted all links in my notes opened through vim plugin going into 'vim' window and not mix with other windows with unrelated stuff.

**Comment by Ram-Z** [comment]
@amerlyq The reason I was interested in this was so I could be able to set up some rules for my WM based on the title. But this didn't seem to work, so I stopped working on this.

I guess it could still be an interesting feature to add. The MR is at #1101. There is no command line option for window names though.

**Comment by The-Compiler** [comment]
@amerlyq That's a nice usecase I didn't consider yet.

What you can do right now is to use a different `--basedir` - however that'll also mean that qutebrowser instance will have a separate config/cache/cookies/etc.

**Comment by wgmayer0** [comment]
Any progress updates?

**Comment by The-Compiler** [comment]
@wgmayer0 If there were, you would see them here in the issue. I'm not awar of anyone working on this.

**Comment by Ram-Z** [comment]
I've been running qb with the changes in https://github.com/qutebrowser/qutebrowser/pull/5446 for a while now. It somewhat addresses this issue too in that the window title now contains `[{session-name}]`. When restricting your sessions to contain a single window, the requested behaviour can be emulated. It still doesn't solve my problem mentioned in https://github.com/qutebrowser/qutebrowser/issues/1096#issuecomment-230091406 though.

**Comment by wgmayer0** [comment]
Ok is there a link somewhere I can follow to implement #5446? Do I need to compile from source?

**Comment by The-Compiler** [comment]
> Currently, all of my qutebrowser windows are named `/`. I would like to have the names of the windows like `firefox` does.

Huh? It doesn't sound like you're talking about the same thing this issue is. I'm not aware of any named window functionality in Firefox.
